corehq/apps/translations/app_translations.py

```diff
@@ -95,60 +93,15 @@ def process_bulk_app_translation_upload(app, workbook):
         # sheet.__iter__ can only be called once, so cache the result
         rows = get_unicode_dicts(sheet)
 
-        # CHECK FOR REPEAT SHEET
-        if sheet.worksheet.title in processed_sheets:
-            msgs.append((
-                messages.error,
-                'Sheet "%s" was repeated. Only the first ' +
-                'occurrence has been processed' %
-                sheet.worksheet.title
-            ))
+        error = _check_for_sheet_error(sheet, expected_sheets=expected_sheets, processed_sheets=processed_sheets)
```

Comment:
this is a great change. one thought is that now that it's extracted to its own function you could manage control flow with exceptions which might be a little more explicit than relying on the return value being `None`

e.g. 
```python
try:
  _check_for_sheet_error(...)
except InvalidSheetException as e:
  msgs.append(e.message)
```

Reply:
810cb075d2f

Reply:
also 72ca8681493